### PR TITLE
remove geojson-pydantic serialization

### LINKED ISSUE #74: remove geojson-pydantic serialization
Geojson-pydantic is used in multiple ways in `tifeatures`:
1. validate the output from the DB https://github.com/developmentseed/tifeatures/blob/master/tifeatures/layer.py#L447
2. defined the output from the baseclass methods https://github.com/developmentseed/tifeatures/blob/master/tifeatures/layer.py#L72 
3. define the output model from the api endpoints https://github.com/developmentseed/tifeatures/blob/master/tifeatures/factory.py#L546

in https://github.com/developmentseed/tifeatures/tree/NoSerialisation we removed geojson-pydantic in all the part to test performance 👇 
```
# NoSerialisation
------------------------------------------------------- benchmark 'Big Feature': 4 tests ------------------------------------------------------
Name (time in ms)                                               Min                   Max                  Mean                Median          
-----------------------------------------------------------------------------------------------------------------------------------------------
test_benchmark_item[html-Saskatchewan]                      37.4572 (1.0)        108.6282 (1.12)        49.3042 (1.0)         38.9626 (1.0)    
test_benchmark_item[geojson-Saskatchewan]                   53.5319 (1.43)        97.0965 (1.0)         67.0153 (1.36)        59.4534 (1.53)   
test_benchmark_item[html-NewfoundlandandLabrador]        1,024.8418 (27.36)    1,209.9585 (12.46)    1,123.8800 (22.79)    1,103.0600 (28.31)  
test_benchmark_item[geojson-NewfoundlandandLabrador]     1,353.3868 (36.13)    1,557.0396 (16.04)    1,431.5393 (29.03)    1,391.4961 (35.71)  
-----------------------------------------------------------------------------------------------------------------------------------------------


# Master
------------------------------------------------------- benchmark 'Big Feature': 4 tests ------------------------------------------------------
Name (time in ms)                                               Min                   Max                  Mean                Median          
-----------------------------------------------------------------------------------------------------------------------------------------------
test_benchmark_item[html-Saskatchewan]                     165.1282 (1.0)        193.6004 (1.0)        181.5587 (1.0)        191.2051 (1.0)    
test_benchmark_item[geojson-Saskatchewan]                  247.0996 (1.50)       276.7573 (1.43)       260.5701 (1.44)       254.2247 (1.33)   
test_benchmark_item[html-NewfoundlandandLabrador]        4,059.6678 (24.58)    4,191.3285 (21.65)    4,117.3781 (22.68)    4,098.8530 (21.44)  
test_benchmark_item[geojson-NewfoundlandandLabrador]     5,949.7388 (36.03)    6,021.0437 (31.10)    5,991.2153 (33.00)    5,987.8045 (31.32)  
-----------------------------------------------------------------------------------------------------------------------------------------------
```
ref: https://gist.github.com/vincentsarago/1e46bd38c4d4bc758631342623144999

### Next 
the output data from postgres is unlikely to be invalid (because we use ST_AsGeojson) so could purely remove **1.**

for **2.**, we could use TypedDict because the only purpose is type checking

For **3.**, we can remove the output model but still add it in openapi https://github.com/developmentseed/tifeatures/blob/NoSerialisation/tifeatures/factory.py#L539

### PR DESCRIPTION
**Pull request by vincentsarago**
closes #74

### CODE REVIEW

File: tests/routes/test_items.py [hunk 1]
```diff
@@ -535,9 +535,9 @@ def test_items_geometry_return_options(app):
     assert len(body["features"]) == 1
     assert body["numberMatched"] == 1
     assert body["numberReturned"] == 1
-    assert body["features"][0]["id"] == "1"
+    assert body["features"][0]["id"] == 1
     assert body["features"][0]["properties"]["ogc_fid"] == 1
-    assert "geometry" not in body["features"][0]
+    assert not body["features"][0]["geometry"]
```
Comment (vincentsarago) [inline tests/routes/test_items.py:540]: We do not exclude `None` so the dict has `geometry=None` (which is still a valid geojson)

File: tests/routes/test_item.py [hunk 2]
```diff
@@ -8,7 +8,7 @@ def test_item(app):
     assert response.headers["content-type"] == "application/geo+json"
     body = response.json()
     assert body["type"] == "Feature"
-    assert body["id"] == "1"
+    assert body["id"] == 1
```
Comment (vincentsarago) [inline tests/routes/test_item.py:11]: it was a bug https://github.com/developmentseed/geojson-pydantic/issues/90

File: tifeatures/factory.py [hunk 3]
```diff
@@ -555,7 +554,8 @@ def queryables(
                         MediaType.json.value: {},
                         MediaType.geojsonseq.value: {},
                         MediaType.ndjson.value: {},
-                    }
+                    },
+                    "model": model.Items,
```
Comment (vincentsarago) [inline tifeatures/factory.py:558]: we add the model in the openapi doc, but it isn't used by the api

File: tifeatures/layer.py [hunk 4]
```diff
@@ -38,6 +37,24 @@
...
+    id: Optional[Any]
+    bbox: Optional[List[float]]
+
+
+class FeatureCollection(TypedDict, total=False):
+    """Simple FeatureCollection model."""
+
+    type: str
+    features: List[Feature]
+    bbox: Optional[List[float]]
```
Comment (vincentsarago) [inline tifeatures/layer.py:55]: Simple Feature and FeatureCollection model for type check purpose only

**Comment by vincentsarago** [comment]
```
## This PR 
--------------------------------------- benchmark 'Items: geojson': 6 tests ----------------------------------------
Name (time in ms)                          Min                 Max                Mean              Median          
--------------------------------------------------------------------------------------------------------------------
test_benchmark_items[geojson-1]         5.8362 (1.0)       12.3128 (1.0)        6.2691 (1.0)        6.0590 (1.0)    
test_benchmark_items[geojson-10]        9.8031 (1.68)      13.0040 (1.06)      10.0499 (1.60)       9.9596 (1.64)   
test_benchmark_items[geojson-50]       26.2651 (4.50)      27.3044 (2.22)      26.7554 (4.27)      26.7215 (4.41)   
test_benchmark_items[geojson-100]      47.2757 (8.10)      77.4294 (6.29)      49.2691 (7.86)      47.6037 (7.86)   
test_benchmark_items[geojson-200]      89.8686 (15.40)    120.8947 (9.82)      93.0947 (14.85)     90.3990 (14.92)  
test_benchmark_items[geojson-250]     112.0345 (19.20)    144.2630 (11.72)    115.9948 (18.50)    112.4964 (18.57)  
--------------------------------------------------------------------------------------------------------------------

## Master 
------------------------------------------ benchmark 'Items: geojson': 6 tests -------------------------------------------
Name (time in ms)                          Min                   Max                  Mean                Median          
--------------------------------------------------------------------------------------------------------------------------
test_benchmark_items[geojson-1]        12.4169 (1.0)         20.3287 (1.0)         14.0450 (1.0)         13.7176 (1.0)    
test_benchmark_items[geojson-10]       48.5490 (3.91)        57.2770 (2.82)        50.6551 (3.61)        49.8315 (3.63)   
test_benchmark_items[geojson-50]      190.7658 (15.36)      195.4025 (9.61)       192.8240 (13.73)      193.3567 (14.10)  
test_benchmark_items[geojson-100]     367.6626 (29.61)      385.0314 (18.94)      376.4546 (26.80)      376.3084 (27.43)  
test_benchmark_items[geojson-200]     756.8030 (60.95)      876.7940 (43.13)      801.7279 (57.08)      776.3115 (56.59)  
test_benchmark_items[geojson-250]     953.5046 (76.79)    1,094.4821 (53.84)    1,012.6114 (72.10)    1,008.9822 (73.55)  
--------------------------------------------------------------------------------------------------------------------------
```

**Review by bitner — APPROVED** [review]
This is all looking great

**Pull request merged**